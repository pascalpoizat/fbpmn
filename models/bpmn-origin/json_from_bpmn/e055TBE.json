{
    "isInterrupt": {
        "BoundaryEvent_02b1vw4": true
    },
    "catN": {
        "StartEvent_1": "NoneStartEvent",
        "Process_1": "Process",
        "BoundaryEvent_02b1vw4": "TimerBoundaryEvent",
        "EndEvent_111i3i3": "NoneEndEvent",
        "Task_0m7n5xk": "AbstractTask",
        "Task_0mfb6xu": "AbstractTask",
        "StartEvent_0hi510e": "NoneStartEvent",
        "SubProcess_0h2vtyo": "SubProcess"
    },
    "catE": {
        "SequenceFlow_19ibfph": "NormalSequenceFlow",
        "SequenceFlow_038z5pz": "NormalSequenceFlow",
        "SequenceFlow_1tuej2e": "NormalSequenceFlow",
        "SequenceFlow_05shv4u": "NormalSequenceFlow",
        "SequenceFlow_1y9pakp": "NormalSequenceFlow"
    },
    "containN": {
        "Process_1": [
            "SubProcess_0h2vtyo",
            "BoundaryEvent_02b1vw4",
            "StartEvent_1",
            "EndEvent_111i3i3"
        ],
        "SubProcess_0h2vtyo": [
            "StartEvent_0hi510e",
            "Task_0mfb6xu",
            "Task_0m7n5xk"
        ]
    },
    "messageE": {},
    "timeInformation": {
        "BoundaryEvent_02b1vw4": {
            "timerDefinitionValue": null,
            "timerDefinitionType": null
        }
    },
    "targetE": {
        "SequenceFlow_19ibfph": "Task_0m7n5xk",
        "SequenceFlow_038z5pz": "Task_0mfb6xu",
        "SequenceFlow_1tuej2e": "EndEvent_111i3i3",
        "SequenceFlow_05shv4u": "SubProcess_0h2vtyo",
        "SequenceFlow_1y9pakp": "Task_0mfb6xu"
    },
    "name": "e055TBE",
    "attached": {
        "BoundaryEvent_02b1vw4": "SubProcess_0h2vtyo"
    },
    "containE": {
        "Process_1": [
            "SequenceFlow_1tuej2e",
            "SequenceFlow_05shv4u"
        ],
        "SubProcess_0h2vtyo": [
            "SequenceFlow_1y9pakp",
            "SequenceFlow_19ibfph",
            "SequenceFlow_038z5pz"
        ]
    },
    "edges": [
        "SequenceFlow_1tuej2e",
        "SequenceFlow_05shv4u",
        "SequenceFlow_1y9pakp",
        "SequenceFlow_19ibfph",
        "SequenceFlow_038z5pz"
    ],
    "messages": [],
    "sourceE": {
        "SequenceFlow_19ibfph": "Task_0mfb6xu",
        "SequenceFlow_038z5pz": "Task_0m7n5xk",
        "SequenceFlow_1tuej2e": "BoundaryEvent_02b1vw4",
        "SequenceFlow_05shv4u": "StartEvent_1",
        "SequenceFlow_1y9pakp": "StartEvent_0hi510e"
    },
    "nodes": [
        "Process_1",
        "SubProcess_0h2vtyo",
        "BoundaryEvent_02b1vw4",
        "StartEvent_1",
        "EndEvent_111i3i3",
        "StartEvent_0hi510e",
        "Task_0mfb6xu",
        "Task_0m7n5xk"
    ],
    "nameN": {}
}