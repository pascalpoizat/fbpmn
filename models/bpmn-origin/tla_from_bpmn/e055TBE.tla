---------------- MODULE e055TBE ----------------

EXTENDS TLC, PWSTypes

VARIABLES nodemarks, edgemarks, net

Interest ==
  "Process_1" :> {  }

ContainRel ==
  "Process_1" :> { "SubProcess_0h2vtyo", "BoundaryEvent_02b1vw4", "StartEvent_1", "EndEvent_111i3i3" }
  @@ "SubProcess_0h2vtyo" :> { "StartEvent_0hi510e", "Task_0mfb6xu", "Task_0m7n5xk" }

Node == { "Process_1", "SubProcess_0h2vtyo", "BoundaryEvent_02b1vw4", "StartEvent_1", "EndEvent_111i3i3", "StartEvent_0hi510e", "Task_0mfb6xu", "Task_0m7n5xk" }

Edge == { "SequenceFlow_1tuej2e", "SequenceFlow_05shv4u", "SequenceFlow_1y9pakp", "SequenceFlow_19ibfph", "SequenceFlow_038z5pz" }

Message == {  }

msgtype ==
  [ i \in {} |-> {}]


source ==
   "SequenceFlow_1tuej2e" :> "BoundaryEvent_02b1vw4"
@@ "SequenceFlow_05shv4u" :> "StartEvent_1"
@@ "SequenceFlow_1y9pakp" :> "StartEvent_0hi510e"
@@ "SequenceFlow_19ibfph" :> "Task_0mfb6xu"
@@ "SequenceFlow_038z5pz" :> "Task_0m7n5xk"

target ==
   "SequenceFlow_1tuej2e" :> "EndEvent_111i3i3"
@@ "SequenceFlow_05shv4u" :> "SubProcess_0h2vtyo"
@@ "SequenceFlow_1y9pakp" :> "Task_0mfb6xu"
@@ "SequenceFlow_19ibfph" :> "Task_0m7n5xk"
@@ "SequenceFlow_038z5pz" :> "Task_0mfb6xu"

CatN ==
   "Process_1" :> Process
@@ "SubProcess_0h2vtyo" :> SubProcess
@@ "BoundaryEvent_02b1vw4" :> TimerBoundaryEvent
@@ "StartEvent_1" :> NoneStartEvent
@@ "EndEvent_111i3i3" :> NoneEndEvent
@@ "StartEvent_0hi510e" :> NoneStartEvent
@@ "Task_0mfb6xu" :> AbstractTask
@@ "Task_0m7n5xk" :> AbstractTask

CatE ==
   "SequenceFlow_1tuej2e" :> NormalSeqFlow
@@ "SequenceFlow_05shv4u" :> NormalSeqFlow
@@ "SequenceFlow_1y9pakp" :> NormalSeqFlow
@@ "SequenceFlow_19ibfph" :> NormalSeqFlow
@@ "SequenceFlow_038z5pz" :> NormalSeqFlow

LOCAL preEdges ==
  [ i \in {} |-> {}]
PreEdges(n,e) == preEdges[n,e]

PreNodes(n,e) == { target[ee] : ee \in preEdges[n,e] }
          \union { nn \in { source[ee] : ee \in preEdges[n,e] } : CatN[nn] \in { NoneStartEvent, MessageStartEvent } }

BoundaryEvent ==
   "BoundaryEvent_02b1vw4" :> [ attachedTo |-> "SubProcess_0h2vtyo", cancelActivity |-> TRUE ]

WF == INSTANCE PWSWellFormed
ASSUME WF!WellTyped
ASSUME WF!WellFormedness

ConstraintNode == TRUE \* none
ConstraintEdge == TRUE \* none
Constraint == TRUE     \* none
INSTANCE PWSConstraints
INSTANCE UserProperties
INSTANCE PWSSemantics

================================================================

