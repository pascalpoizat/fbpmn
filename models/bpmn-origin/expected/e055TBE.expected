Working in /tmp/e055TBE.xOaFe with 1 worker(s)
[92mtransformation done[0m
<<"Processes=", 1, "Nodes=", 8, "Gateway=", 0, "SF=", 5, "MF=", 0>>
---------- Network01Bag ----------
[X] Prop01Type
     states=10 trans=16 depth=8
[X] Prop02Safe
     states=10 trans=16 depth=8
[X] Prop03Sound
     states=10 trans=16 depth=8
[X] Prop04MsgSound
     states=10 trans=16 depth=8
---------- Network02FifoPair ----------
[X] Prop01Type
     states=10 trans=16 depth=8
[X] Prop02Safe
     states=10 trans=16 depth=8
[X] Prop03Sound
     states=10 trans=16 depth=8
[X] Prop04MsgSound
     states=10 trans=16 depth=8
---------- Network03Causal ----------
[X] Prop01Type
     states=10 trans=16 depth=8
[X] Prop02Safe
     states=10 trans=16 depth=8
[X] Prop03Sound
     states=10 trans=16 depth=8
[X] Prop04MsgSound
     states=10 trans=16 depth=8
---------- Network04Inbox ----------
[X] Prop01Type
     states=10 trans=16 depth=8
[X] Prop02Safe
     states=10 trans=16 depth=8
[X] Prop03Sound
     states=10 trans=16 depth=8
[X] Prop04MsgSound
     states=10 trans=16 depth=8
---------- Network05Outbox ----------
[X] Prop01Type
     states=10 trans=16 depth=8
[X] Prop02Safe
     states=10 trans=16 depth=8
[X] Prop03Sound
     states=10 trans=16 depth=8
[X] Prop04MsgSound
     states=10 trans=16 depth=8
---------- Network06Fifo ----------
[X] Prop01Type
     states=10 trans=16 depth=8
[X] Prop02Safe
     states=10 trans=16 depth=8
[X] Prop03Sound
     states=10 trans=16 depth=8
[X] Prop04MsgSound
     states=10 trans=16 depth=8
---------- Network07RSC ----------
[X] Prop01Type
     states=10 trans=16 depth=8
[X] Prop02Safe
     states=10 trans=16 depth=8
[X] Prop03Sound
     states=10 trans=16 depth=8
[X] Prop04MsgSound
     states=10 trans=16 depth=8
done.
